src/main/java/com/puppycrawl/tools/checkstyle/checks/SuppressWarningsHolder.java

```diff
@@ -522,6 +537,18 @@ private static String getStringExpr(DetailAST ast) {
         return valueList;
     }
 
+    /**
+     * Remove preceding newline and whitespace from the content of a text block.
+     *
+     * @param textBlockContent the actual text in a text block.
+     * @return content of text block with preceding whitespace and newline removed.
+     */
+    private static String getContentWithoutPrecedingWhitespace(String textBlockContent) {
+        final String contentWithNoPrecedingNewline =
+            NEWLINE.matcher(textBlockContent).replaceAll("");
+        return WHITESPACE.matcher(contentWithNoPrecedingNewline).replaceAll("");
```

Comment:
I wonder about the behavior when the text block has more than just leading/trailing whitespace (or, does this match what the check would do for a normal string literal in that case?)

Reply:
I agree. We need a helper method to extract the content of a text block in the same way as the compiler does. This may be a bit tricky.
@nmancus1 is this the only check where such method should be used? If not, the method should be added to CheckUtil or CommonUtil (in a separate issue).

Reply:
Done.

Reply:
Fixed.